api/src/main/java/io/druid/timeline/partition/NoneShardSpec.java

```diff
@@ -32,6 +33,11 @@
  */
 public class NoneShardSpec implements ShardSpec
 {
+  private final static NoneShardSpec INSTANCE = new NoneShardSpec();
```

Comment:
@xvrl : I thought about it earlier but was left it public in order to not break any custom extensions using it with a minor version update in druid. If you feel strongly about it, let me know, i can make the change.